File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        int[] answer = {};
        Stack<Integer> stack = new Stack<>();
        
        for(int i = 0; i < arr.length; i++){
            int num = arr[i];
            int len = flag[i] ? num * 2 : num;
            for(int j = 0; j < len; j++){
                if(flag[i]) {
                    stack.push(num);
                }else {
                    stack.pop();
                }
            }
        }
        
        answer = new int[stack.size()];
        
        for(int k = answer.length - 1; k >= 0; k--) answer[k] = stack.pop();
        
        
        return answer;
    }
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    answer = []
    
    for i in range(len(flag)):
        X = arr[i]
        if flag[i]:
            for n in range(0, X * 2):
                answer.append(X)
        else:
            for m in range(0, X):
                answer.pop()

            
    
    return answer